### DISCUSSION TITLE
Basic Table Operations

### DISCUSSION THREAD

**Discussion by ghislaineguerin** [mathesar-foundation/mathesar#357] [Maintainer discussions]
The specs related to issue #185  are ready for review. Please take a look at the attached document and follow the [review guidelines](https://wiki.mathesar.org/en/design/process/review-guidelines).

## Document Link
https://wiki.mathesar.org/en/design/specs/table-operations

@mathemancer @kgodey @pavish, your feedback is required.
The deadline for feedback is set to a week from now, the first review round will be completed by Friday 16th. Please plan your review so that at least one round of changes can be made and reviewed if needed.

**Comment by kgodey** [comment]
This looks good to me. Nice work, @ghislaineguerin!

Some feedback and questions:
(1) The confirmation prompt for deleting a table should have space for showing a list of related tables (and later views).
(2) Is "Duplicate Table" meant to be implemented now?
(3) It might be nice to add some text next to the "..." menu to make it more obvious to users.
(4) I don't see the interaction for editing an existing table name. Can you point to it?

**Reply by ghislaineguerin** [reply]
> (1) The confirmation prompt for deleting a table should have space for showing a list of related tables (and later views).

Agreed, will update the prototype to reflect this.

> (2) Is "Duplicate Table" meant to be implemented now?

No, it was to show that other table-related options might go there, I will point this out in the implementation notes.

> (3) It might be nice to add some text next to the "..." menu to make it more obvious to users.

Makes sense. I'll add something.

> (4) I don't see the interaction for editing an existing table name. Can you point to it?

I've missed adding the interaction. Click on any of the name labels on a saved table.

**Comment by mathemancer** [comment]
I like it!  I only have a clarifying question:
Is the user able to actually _save_ the table with the placeholder name, or is it truly a placeholder that needs to be filled in before saving?
   - This is relevant to know, since that implies the front end checks the currently-existing tables before choosing the placeholder name (or that the backend actually chooses the placeholder name, and the frontend asks for it).

**Reply by ghislaineguerin** [reply]
I think the goal would be to provide a name that the user can save if they aren't ready to name a table. So yes, they'd actually save with the provided name if they leave it unchanged.

**Reply by mathemancer** [reply]
Okay, thank you!  I'll take that into account.

**Comment by pavish** [comment]
Looks good to me!

**Comment by ghislaineguerin** [comment]
@pavish @kgodey @mathemancer 
Review notes added to the doc and posted below for reference:
Changes introduced to the prototype:
- Location of the table options menu
- Delete dialog updated with a list of related tables

## Placeholder Names
Naming tables should be optional at the moment of creation. The user should get a naming suggestion every time they create a new table. Mathesar should have the ability to auto-generate such names with incremental numbers. The decision to do so in the front-end or back-end will be decided later.

## Table Options Menu
A menu that contains table-related options can be accessed from the top toolbar. This menu should have additional functionality in the future, such as 'Duplicate Table,' which is only included now for reference.

## Showing Related Tables and Views in the Delete Dialog
When a user decides to delete a table, they will have to review a list of all related tables and views before confirming their choice.
